src/components/common/PrivateRoute.js

```diff
@@ -1,67 +1,19 @@
-import React, { useState, useEffect } from 'react';
-import { connect } from 'react-redux';
-import { useHistory } from 'react-router-dom';
-import { SecureRoute, useOktaAuth } from '@okta/okta-react';
-import { authenticateUser } from '../../state/actions/auth/authenticateUser';
-import { getProfile } from '../../state/actions/userProfile/getProfile';
+import React from 'react';
+import { Route } from 'react-router-dom';
+import { withAuthenticationRequired } from '@auth0/auth0-react';
 import Sidebar from './Sidebar/Sidebar';
-import LoadingComponent from './LoadingComponent';
 
-const PrivateRoute = ({
-  component: Component,
-  path,
-  redirect,
-  allowRoles, // should be an array of allowed role_id's i.e. [3, 4]
-  isAuthenticated,
-  profile_id,
-  userProfile,
-  dispatch,
-  ...rest
-}) => {
-  const { push } = useHistory();
-  const { authState, oktaAuth } = useOktaAuth();
-  const [loading, setLoading] = useState(true); // hiding contents
-  useEffect(() => {
-    if (Object.keys(userProfile).length === 0) {
-      if (profile_id === null) {
-        if (authState !== null && authState.isAuthenticated) {
-          dispatch(authenticateUser(authState, oktaAuth));
-        } else {
-          push(redirect);
-        }
-      } else {
-        dispatch(getProfile(profile_id));
-      }
-    } else if (allowRoles.includes(userProfile.role_id)) {
-      setLoading(false);
-    } else {
-      push(redirect);
-    }
-  }, [
-    isAuthenticated,
-    userProfile,
-    profile_id,
-    allowRoles,
-    redirect,
-    dispatch,
-    oktaAuth,
-    authState,
-    push,
-  ]);
+const PrivateRoute = ({ component, path, ...args }) => (
+  <Sidebar>
+    <Route
+      path={path}
+      /* NOTE: this is commented out for current functionality, but is a template for building auth0 into the app*/
```

Comment:
Great use of code comments. Not only informational but also showing the current use and future implementation is fantastic. Good job!